modular_skyrat/modules/vore/code/obj/belly.dm

```diff
@@ -0,0 +1,304 @@
+		return
+	for (var/varname in _data)
+		if (!(varname in static_belly_vars()))
+			_data -= varname
+	var/static/default_belly_list = default_belly_info()
+	for (var/varname in default_belly_list)
+		if (!(varname in _data))
+			_data[varname] = default_belly_list[varname]
+	data = _data
+	name = data[BELLY_NAME]
+	desc = data[BELLY_DESC]
+	mode = data[BELLY_MODE]
+	can_taste = data[BELLY_CAN_TASTE] == "Yes" ? TRUE : FALSE
+	swallow_verb = data[BELLY_SWALLOW_VERB]
+	check_mode()
+	if (!isnull(bellynum))
+		belly_ref = bellynum
+
+//people don't want to have to wear internals while inside someone
+//todo: should REALLY find a better way to do this
+/obj/vbelly/assume_air(datum/gas_mixture/giver)
+	var/datum/gas_mixture/copy = air_contents.copy()
+	return copy.merge(giver)
+/obj/vbelly/return_air()
+	var/datum/gas_mixture/copy = air_contents.copy()
+	return copy
+/obj/vbelly/return_analyzable_air()
+	var/datum/gas_mixture/copy = air_contents.copy()
+	return copy
+/obj/vbelly/remove_air(amount)
+	var/datum/gas_mixture/copy = air_contents.copy()
+	return copy.remove(amount)
+
+/obj/vbelly/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	. = ..()
+	if (isliving(arrived))
+		var/mob/living/arrived_mob = arrived
+		if (desc && arrived_mob.check_vore_toggle(SEE_OTHER_MESSAGES, VORE_CHAT_TOGGLES))
+			to_chat(arrived_mob, span_notice(desc))
+		var/datum/component/vore/vore = arrived_mob.GetComponent(/datum/component/vore)
+		if (owner.check_vore_toggle(SEE_OTHER_MESSAGES, VORE_CHAT_TOGGLES) && can_taste)
+			to_chat(owner, span_notice("[arrived_mob] tastes of [vore.char_vars["tastes_of"]]."))
+		RegisterSignal(arrived, COMSIG_LIVING_RESIST, .proc/prey_resist)
+		check_mode()
+		arrived_mob.become_blind(belly_string_ref)
+	if (iscarbon(arrived))
+		var/mob/living/carbon/carbon_arrived = arrived
+		carbon_arrived.update_suit_sensors()
+	update_static_vore_data()
+
+/obj/vbelly/Exited(atom/movable/gone, direction)
+	. = ..()
+	if (gone in absorbed)
+		absorbed -= gone
+		UnregisterSignal(gone, COMSIG_PARENT_EXAMINE)
+	if (gone in absorbing)
+		absorbing -= gone
+	if (isliving(gone))
+		UnregisterSignal(gone, COMSIG_LIVING_RESIST)
+		var/mob/living/living_gone = gone
+		living_gone.cure_blind(belly_string_ref)
+		if (living_gone.client?.prefs?.vr_prefs)
+			living_gone.client.prefs.vr_prefs.needs_update |= UPDATE_INSIDE
+			living_gone.client.prefs.vr_prefs.update_static_data(living_gone)
+	if (iscarbon(gone))
+		var/mob/living/carbon/carbon_gone = gone
+		carbon_gone.update_suit_sensors()
+	check_mode()
+	update_static_vore_data()
+
+/obj/vbelly/proc/update_static_vore_data(force=FALSE, only_contents=FALSE, timer = FALSE)
+	static_data_cooldown = (force || timer) ? 0 : static_data_cooldown
+	static_timer = timer ? FALSE : static_timer
+	if (static_data_cooldown > world.time)
+		if (!static_timer)
+			static_timer = TRUE
+			addtimer(CALLBACK(src, .proc/update_static_vore_data, FALSE, FALSE, TRUE), static_data_cooldown - world.time)
```

Comment:
This seems incredibly hacky and like something isn't being done right to tgui spec.